irods/prc_write_irodsA.py

```diff
@@ -0,0 +1,50 @@
+import getopt
+import getpass
+import textwrap
+import sys
+
+from irods.auth.pam_password import _get_pam_password_from_stdin as get_password
+from irods.client_init import (
+    write_pam_irodsA_file,
+    write_native_irodsA_file
+)
+
+if __name__ == "__main__":
+    extra_help = textwrap.dedent(
+        """
+    This Python module also functions as a script to produce a "secrets" (i.e. encoded password) file.
+    Similar to iinit in this capacity, if the environment - and where applicable, the PAM
+    configuration for both system and user - is already set up in every other regard, this program
+    will generate the secrets file with appropriate permissions and in the normal location, usually:
+
+       ~/.irods/.irodsA
+
+    The user will be interactively prompted to enter their cleartext password.
+    """
+    )
+
+    vector = {
+        "pam_password": write_pam_irodsA_file,
+        "native": write_native_irodsA_file
+    }
+    opts, args = getopt.getopt(sys.argv[1:], "hi:", ["ttl=","help"])
+    optD = dict(opts)
+    help_selected = {*optD} & {'-h','--help'}
+    if len(args) != 1 or help_selected:
+        print("{}\nUsage: {} [-i STREAM| -h | --help | --ttl HOURS] AUTH_SCHEME".format(extra_help, sys.argv[0]))
+        print("  Choices for AUTH_SCHEME are:")
+        for x in vector:
+            print("    {}".format(x))
+        print("  STREAM is the name of a file containing a password. Alternatively, a hyphen('-') is used to\n"
+              "  indicate that the password may be read from stdin.")
+        sys.exit(0 if help_selected else 1)
+    elif args[0] in vector:
+        options = {}
+        inp_stream = optD.get('-i',None)
+        if '--ttl' in optD:
+            options['ttl'] = optD['--ttl']
+        pw = get_password(sys.stdin if inp_stream in ("-",None)
+                          else open(inp_stream,"r"))
+        vector[args[0]](pw,**options)
+    else:
+        print("did not recognize authentication scheme argument", file=sys.stderr)
```

Comment:
Is this identical code that originally lived in client_init.py?

Reply:
Near-identical, yes.   I think i may have put in some changes. If those have been squashed out , I could go and look for the original commits.

Reply:
No need. As long as the code is tested and confirmed to work as intended, we're good.

Reply:
It should probably have a thorough test dedicated to it.

Reply:
Agreed.

Reply:
It is used once in `test007_pam_features_in_new_auth_framework.bats` along with the --ttl option.  Think that's almost sufficient .  I'll make a simple `test008*.bats` that uses it to write a new .irodsA for native auth.

Reply:
I think the tests are adequate  now, so I will resolve.